src/Badge.tsx

```diff
@@ -21,12 +21,20 @@ export interface IBadge extends IBox<HTMLSpanElement> {
 export const Badge: React.FunctionComponent<IBadge> = props => {
   const { color = BadgeColor.Default, variant = BadgeVariant.Pill, ...restProps } = props;
 
-  return <Box {...restProps} as="span" css={badgeStyles({ color, variant })} />;
-};
-
-const badgeStyles = ({ color, variant }: { color: BadgeColor; variant: BadgeVariant }) => {
   const { badge: theme } = useTheme();
```

Comment:
Any reason for moving `useTheme` to the body of component?

Reply:
Yup! Using hooks in functions that are not either A) a react component or B) another hook is considered an anti-pattern (as far as I can tell). This makes sense, as it is all too easy to mistakenly invoke a hook conditionally if it is not easy to tell where it is being used. We made this mistake several times (see blockquote and context menu for a couple of examples). By sticking to using hooks in those two spots, it will be more obvious if it is being used conditionally.

The `tslint-react-hooks` ruleset I added checks this for us.

Reply:
> This makes sense, as it is all too easy to mistakenly invoke a hook conditionally if it is not easy to tell where it is being used.

Ok, you convinced me :)

Usage of `useTheme` in blockQuote was actually alright, but agree about ContextMenu and Menu. 

Actually, I was trying to point that conditional usage a couple of times, but it's still difficult to spot in all places, especially when only a part of the code (diff) is displayed, so it's great to have a linter rule for that. Also, verifying that in the app is not that trivial either, because you usually need to meet some condition / get to some state.
I committed that mistake once or twice as well, so yet again very happy to have a linter do the job for us.

Reply:
Ah, I meant `Dialog.jsx` - original pr description updated! (I discovered this issue via Studio bug).

Reply:
Really? I believe I did place it before the `(show)` condition and tested it 'live' a couple of time in Storybook (show on/off).